packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -33,6 +30,25 @@ export class ResourceFactory {
 
     public makeParams() {
         return {
+            [ResourceConstants.PARAMETERS.AppSyncApiName]: new StringParameter({
+                Description: 'The name of the AppSync API',
+                Default: 'AppSyncSimpleTransform'
+            }),
+            [ResourceConstants.PARAMETERS.APIKeyExpirationEpoch]: new NumberParameter({
+                Description: 'The epoch time in seconds when the API Key should expire.' +
+                    ' Setting this to 0 will default to 180 days from the deployment date.' +
+                    ' Setting this to -1 will not create an API Key.',
+                Default: 0,
+                MinValue: -1
+            }),
+            [ResourceConstants.PARAMETERS.CreateAPIKey]: new NumberParameter({
+                Description: 'The boolean value to control if an API Key will be created or not.' +
```

Comment:
IMHO using the epoch for two different things, specifying absolute expiration and disabling creation of API key is confusing for customers, and since now we've a way of configuring API key expiration during the service walkthrough and since API key creation depends on the configured auth providers I added this specific property to control this and also marked with a warning ```APIKeyExpirationEpoch``` deprecated, but we're translating it into CreateAPIKey when possible.